src/backend/api/meals.js

```diff
@@ -2,13 +2,125 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+router.get("/", async (req, res) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const meals = knex("homework3.Meal");
+
+    const {
+      maxPrice,
+      availableReservations,
+      title,
+      dateAfter,
+      dateBefore,
+      limit,
+      sortKey,
+      sortDir
+    } = req.query;
+
+    if (maxPrice !== undefined) {
+      meals = meals.where("price", "<", maxPrice);
+    }
+
+    if (availableReservations !== undefined) {
+      if (availableReservations === "true") {
+        meals = meals.whereRaw("max_reservations > reservations");
+      } else if (availableReservations === "false") {
```

Comment:
Two questions: 

1) What if `availableReservations` is neither of those values? Do you think it will be necessary to have a fallback scenario for that?

2) Can you minimize this if-else block using the conditional operator? (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator)

Reply:
Thanks for the suggestions, now it looks much more pro 😁